build.gradle

```diff
@@ -62,6 +63,12 @@ dependencies {
         implementation(group: 'org.codehaus.jettison', name: 'jettison', version: '1.5.4')
     }
 
+    dependencyManagement {
+        dependencies {
+            dependency group: 'org.springframework', name: 'spring-beans', version: '5.3.31-wso2v1'
```

Comment:
There is no property that can be used to override `spring-beans` version.
Ref - https://docs.spring.io/spring-boot/docs/current/reference/html/dependency-versions.html#appendix.dependency-versions.properties

There is `spring-framework.version` available but this will override versions of all spring boot managed dependencies. Upgrading those might cause BD to report new vulnerabilities in the project.

We can not use gradle constraints here because spring dependency management plugin's resolution strategy triumphs any such constraints. Which is why we had to use version properties to upgrade spring-* versions earlier (in #1).
Ref- https://github.com/spring-projects/spring-boot/issues/26840#issuecomment-857865885

Now, one more concern would be whether we are introducing direct dependency here. 
The dependencyManagement does not do this. Verified by going through dependency tree (`gradlew dependencies` task)

Reply:
I'm ok with this.  However, we should always consider how changes like this impact future version increases for all dependencies, to make sure we will not add complexity or cause any hidden conflicts or incompatibilities.  For example, when we upgrade the spring version in future, will this library be held back to this specific version then?  Can we constraint it to say `5.3.31-wso2v1+` ("or a newer version")?